ZFSin/spl/include/sys/vnode.h

```diff
@@ -416,6 +416,8 @@ static inline int win_has_cached_data(struct vnode *vp)
 	} while(0)
 #else
 #define vnode_pager_setsize(vp, sz)  do { \
+		KIRQL OldIrql; \
+		KeAcquireSpinLock(&vp->v_spinlock, &OldIrql); \
```

Comment:
Rather than accessing internals of vnode, it should probably vnode_get(vp), to grab an iocount, test return code for success, and VN_RELE when done with it. Inside vnode_get we do the spinlock and make sure it isn't released.